[Routing] remove deprecations for 3.0

Tobion opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | no, but unrelated |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Tobion committed:
[Routing] remove deprecated features from routing

Tobion committed:
[FrameworkBundle] remove deprecated routing features

Tobion commented:
@fabpot ready

Tobion commented:
I forgot `YamlFileLoader::availableKeys`

Tobion commented:
Fixed

fabpot commented:
Looks there is a problem somewhere as the tests do not pass.

Tobion committed:
[FramworkBundle] our tests now expect symfony 3.0 behavior for routing

Tobion commented:
@fabpot I had to update the routing requirement in the frameworkbundle. The component tests should now succeed when the PR is merged.

fabpot commented on src/Symfony/Bundle/FrameworkBundle/composer.json:
```diff
@@ -23,7 +23,7 @@
         "symfony/http-kernel": "~2.7|~3.0",
         "symfony/filesystem": "~2.7|~3.0",
-        "symfony/routing": "~2.7|~3.0",
+        "symfony/routing": "~3.0",
```
That would be the first where this would be needed.

nicolas-grekas replied:
Looks like this is not the root of the pb: components=low are still failing but subtree-splitting won't fix them. Maybe you need to add a higher req for one of the 2nd level deps? See <2.7 vers:
https://travis-ci.org/symfony/symfony/jobs/47218155#L3011

nicolas-grekas replied:
Ok, I'm wrong...

Tobion replied:
It's only required for tests not the actual functionality (because _scheme/_method would not hurt). Is it possible to have the same dependecy with different versions for `require` and `require-dev`?
I.e. `~2.7|~3.0` in `require` but `~3.0` in `require-dev`?

nicolas-grekas replied:
No it's not unfortunately. You should adjust the tests to make them less strict. With [assertStringMatchesFormat](https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertStringMatchesFormat) it should be possible.

Tobion replied:
Or we just leave it this way. Sooner or later its gonna happen anyway.

stof replied:
why is it required for tests of the FrameworkBundle ? Which test is actually breaking when running it on Routing 2.7 ?

Tobion replied:
the one describing routes in the console in xml, yaml etc format

Tobion commented:
@symfony/deciders ping

stof commented:
:+1:

Tobion committed:
[Routing] make path required again in the xsd

pattern was previously also required, see https://github.com/symfony/symfony/commit/508299400d7fe6f6222a32d5e527d988bb307dcb#diff-4

fabpot commented:
I'm not satisfied with the Composer dep as everywhere else we were able to avoid doing that.

Tobion commented:
Well otherwise we would need to add the `unset($requirements['_scheme'], $requirements['_method']);` again. But as we want to get rid of legacy code in 3.0 I don't think that's what we want.

Tobion commented:
And just fixing the test would still confuse users. Because they would see a `_scheme` requirement in the description even when they didn't even use the deprecated definition.
